File: app/src/main/java/org/sil/storyproducer/model/TemplateSlide.java
package org.sil.storyproducer.model;

import org.sil.storyproducer.tools.media.graphics.KenBurnsEffect;

import java.io.File;

/**
 * This class contains metadata pertinent to a given slide from a story template.
 */
public class TemplateSlide {

    private File mNarrationAudio;
    private File mImage;
    private KenBurnsEffect mKBFX;
    private File mSoundtrack;
    private int mSoundtrackVolume;

    public TemplateSlide(File narrationAudio, File image, KenBurnsEffect kbfx, File soundtrack, int soundtrackVolume) {
        mNarrationAudio = narrationAudio;
        mImage = image;
        mKBFX = kbfx;
        mSoundtrack = soundtrack;
        mSoundtrackVolume = soundtrackVolume;
    }

    public File getNarrationAudio() {
        return mNarrationAudio;
    }

    public File getImage() {
        return mImage;
    }

    public KenBurnsEffect getKenBurnsEffect() {
        return mKBFX;
    }

    public File getSoundtrack() {
        return mSoundtrack;
    }

    public int getSoundtrackVolume() {
        return mSoundtrackVolume;
    }
}

File: app/src/main/java/org/sil/storyproducer/model/TemplateSlide.kt
package org.sil.storyproducer.model

import android.graphics.Rect

import org.sil.storyproducer.tools.media.graphics.KenBurnsEffect

import java.io.File

/**
 * This class contains metadata pertinent to a given slide from a story template.
 */
data class TemplateSlide(
        val narrationAudio: File?,
        val image: File?,
        val imageDimensions: Rect,
        val kenBurnsEffect: KenBurnsEffect,
        val soundtrack: File?,
        val soundtrackVolume: Int)
